### ISSUE TITLE
Error building DATASET and print heatmaps annotations in PoseC3D

### ISSUE THREAD

**Issue by christina05011** [open-mmlab/mmaction2#2560]
### Branch

main branch (1.x version, such as `v1.0.0`, or `dev-1.x` branch)

### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmaction2/issues) and [Discussions](https://github.com/open-mmlab/mmaction2/discussions) but cannot get the expected help.
- [X] I have read the [documentation](https://mmaction2.readthedocs.io/en/latest/) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmaction2).

### Environment

I have this versions:

- mmengine 0.7.4
- mmcv 2.0.0
- mmdet 3.0.0
- mmpose 1.0.0
- mmaction2 1.0.0

### Describe the bug

Hi, I have a problem when I am building datasets in PoseC3D. 

First, I did the steps according to the [tutorial ](https://colab.research.google.com/github/open-mmlab/mmaction2/blob/master/demo/mmaction2_tutorial.ipynb#scrollTo=dDBWkdDRk6oz), but I want to print only heatmaps annotations.

### Reproduces the problem - code sample

I run:

```
dataloader_cfg = copy.deepcopy(cfg.train_dataloader)
dataset_cfg = dataloader_cfg.pop('dataset')
dataset = DATASETS.build(dataset_cfg)
print(next(iter(dataset)))
```

### Reproduces the problem - error message

When I run those commands with config tsn_imagenet-pretrained-r50_8xb32-1x1x3-100e_kinetics400-rgb.py, I have results resized and not empty. 

![image](https://github.com/open-mmlab/mmaction2/assets/42581109/0d314823-91be-495c-a59f-8193334d2b63)

But when I run with config slowonly_r50_8xb16-u48-240e_ntu60-xsub-keypoint.py, I only have zeros. Also, when I train, I only have zeros too. 

![image](https://github.com/open-mmlab/mmaction2/assets/42581109/6940b6c9-bfda-42bc-9cb9-2ffaca507cbe)

### Additional information

- How can I print only heatmaps annotations? Did I do it correctly?
- Why does it only print zeros (none data)?

Thank you for the answers.

**Comment by rodp63** [comment]
Hello there, same problem here. Do you have any update on this? Thank you!

**Comment by christina05011** [comment]
Hi @cir7. Do you have any update on this? Thanks!

**Comment by cir7** [comment]
Could you share the config file you are using for training? And are you using a custom dataset?

**Comment by christina05011** [comment]
Hi @cir7, thank you for the answer.

I use this [config file](https://github.com/open-mmlab/mmaction2/blob/main/configs/skeleton/posec3d/slowonly_r50_8xb16-u48-240e_ntu60-xsub-keypoint.py), but I changed the numbers of workers to 2 and the batch size to 1 in dataloaders. I made this configurations for one GPU in Colab like the [tutorial](https://colab.research.google.com/github/open-mmlab/mmaction2/blob/master/demo/mmaction2_tutorial.ipynb#scrollTo=dDBWkdDRk6oz).

And, I use the same .pkl file [ntu60_2d.pkl](https://github.com/open-mmlab/mmaction2/blob/main/tools/data/skeleton/README.md).

**Comment by cir7** [comment]
Ok, I found that the heatmap you got is reasonable. PoseC3D uses a special heatmap annotation, which maps the keypoints to an all-zero image according to the spatial coordinates, so most values of the input are zeros, except for the keypoints, this [visualization](https://user-images.githubusercontent.com/34324155/116529336-6dff8d00-a90f-11eb-807e-4d9168997655.gif) shows the sparsity. You could refer to the paper for mode details. While the general 3D models take rgb images as input, which has a mean value of 128.

**Comment by christina05011** [comment]
Thanks for the reply @cir7.

I check the output and I have the keypoints (non-zero data). But, if I only want the heatmap annotations, is it correct to print with DATASETS.build() (like I run) or with Runner.build_dataloader(dataloader=train_dataloader_cfg)?

**Comment by cir7** [comment]
1. dataloader stack batch of samples, which could add a new dim, while dataset returns a single sample 
2. You could easily ignore the first dim of shape `[1, 17, 48, 56, 56]`, different `T` dim (32 vs 48) means  different sample strategies, the paper takes both for experiments.

**Comment by christina05011** [comment]
Thanks @cir7 for the answers. You solved my doubts!
